sdk/core/azure-core/inc/uuid.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <new> // for placement new
+#include <string>
+#include <utility> // for swap and move
+#include <random>
+
+namespace Azure { namespace Core {
+
+  class UUID {
+
+  private:
+    static const int UUIDSize = 16;
```

Comment:
(suggestion) `static const int` -> `constexpr auto`

Reply:
I don't feel it's necessary to make it `auto`. After all, we don't have a rule that NO variable types at all, right?

Reply:
No rule at all, especially in this case I'd leave to the individual preference. But the type of int literal without any suffixes is int anyway, and the code that uses it really does not care it to be exactly `int`, just some integral type... Plus having it as constexpr almost makes it a typeless numeric literal. I don't know, it that was me I would be using autos and decltypes everywhere :)
Really a minor thing to me, I am totally fine with not using auto here (if that was `auto x = f()`, there sometimes more reasons to argue for it).
Unless I am missing something. Why would you prefer `int` instead of `auto` in cases like this, do you have an example when it causes potential problems / or have past experience with this when it caused problems?

Reply:
@antkmsft just personal preference. I feel it's hard to tell the type when we write `auto x = f()`, especially when `f()` returns a user-defined struct. Anyway, it's not a big issue.

Reply:
I think that with auto you would run into less problems when maintaining the code.
Return types may change to return derived classes, but then suddenly your code that used to copy does slicing to a base class. I agree that in some cases, it is a small inconvenience to the code reader to not see the concrete type.